### PR TITLE
feat: add svg plugin

### PR DESCRIPTION
**Pull request by asalzburger**
This PR adds a new plugin for displaying geometry based on the [ACTSVG](https://github.com/acts-project/actsvgl) library.
Here are some first plots, the ACTSVG library is not yet capable of dealing with plane layer types, which is why this is not yet supported.

Screenshot 2022-08-02 at 16 52 03

Screenshot 2022-08-02 at 16 52 27

Screenshot 2022-08-02 at 16 52 49

sheet_annulus

odd_vol_16_layer_6_grid

odd_vol_16_layer_6_modules

### CODE REVIEW

File: Plugins/ActSVG/include/Acts/Plugins/ActSVG/LayerSvgConverter.hpp [hunk 1]
```diff
@@ -0,0 +1,47 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2021 CERN for the benefit of the Acts project
```
Comment (AJPfleger) [inline Plugins/ActSVG/include/Acts/Plugins/ActSVG/LayerSvgConverter.hpp:3]: Here and in other files we have the wrong year (5x2021, 2x2020), I think.

Comment (asalzburger) [reply Plugins/ActSVG/include/Acts/Plugins/ActSVG/LayerSvgConverter.hpp:3]: Oh, thanks! will check.

**Comment by asalzburger** [comment]
Hi @paulgessinger - how would you go about the failing tests, I think best would be to include `ACTS_BUILD_PLUGIN_ACTSVG` into build everything?

File: Examples/Io/CMakeLists.txt [hunk 2]
```diff
@@ -6,3 +6,4 @@ add_subdirectory(NuclearInteractions)
 add_subdirectory(Obj)
 add_subdirectory(Performance)
 add_subdirectory(Root)
+add_subdirectory_if(Svg ACTS_BUILD_PLUGIN_ACTSVG)
```
Comment (AJPfleger) [inline Examples/Io/CMakeLists.txt:9]:
```suggestion
add_subdirectory_if(Svg ACTS_BUILD_PLUGIN_ACTSVG)
```

File: Examples/Python/src/JsonStub.cpp [hunk 3]
```diff
@@ -9,5 +9,5 @@
 #include "Acts/Plugins/Python/Utilities.hpp"
 
 namespace Acts::Python {
-void addJson(Context& ctx) {}
+void addJson(Context& /*unused*/) {}
 }  // namespace Acts::Python
```
Comment (AJPfleger) [inline Examples/Python/src/JsonStub.cpp:13]:
```suggestion
}  // namespace Acts::Python
```

File: Examples/Python/src/SvgStub.cpp [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "Acts/Plugins/Python/Utilities.hpp"
+
+namespace Acts::Python {
+void addSvg(Context& /*unused*/) {}
+}  // namespace Acts::Python
```
Comment (AJPfleger) [inline Examples/Python/src/SvgStub.cpp:13]:
```suggestion
}  // namespace Acts::Python
```

File: thirdparty/actsvg/README.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+# Build Recipe for ActSVG
+
+This directory holds a simple build recipe for the
+[VecMem](https://github.com/acts-project/actsvg) project. Used in case
+`ACTS_USE_SYSTEM_ACTSVG` is set to `FALSE` for the build.
```
Comment (AJPfleger) [inline thirdparty/actsvg/README.md:5]:
```suggestion
[VecMem](https://github.com/acts-project/actsvg) project. It is used in
case `ACTS_USE_SYSTEM_ACTSVG` is set to `FALSE` for the build.
```

File: Tests/UnitTests/Plugins/ActSVG/TrackingGeometrySvgConverterTests.cpp [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+  auto geometrySheets = Acts::Svg::TrackingGeometryConverter::convert(
+      tgContext, *tGeometry, tgOptions);
+
+  for (const auto& s : geometrySheets) {
+    Acts::Svg::toFile({s}, s._id + ".svg");
+  }
+}
+
+BOOST_AUTO_TEST_SUITE_END()
```
Comment (AJPfleger) [inline Tests/UnitTests/Plugins/ActSVG/TrackingGeometrySvgConverterTests.cpp:61]:
```suggestion
BOOST_AUTO_TEST_SUITE_END()
```

File: Tests/UnitTests/Plugins/ActSVG/LayerSvgConverterTests.cpp [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+          Acts::Svg::toFile({s}, s._id + ".svg");
+        }
+      }
+    }
+  }
+}
+
+BOOST_AUTO_TEST_CASE(PlaeyLayerSvg) {}
+
+BOOST_AUTO_TEST_SUITE_END()
```
Comment (AJPfleger) [inline Tests/UnitTests/Plugins/ActSVG/LayerSvgConverterTests.cpp:219]:
```suggestion
BOOST_AUTO_TEST_SUITE_END()
```

File: Plugins/ActSVG/src/TrackingGeometrySvgConverter.cpp [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+    }
+  }
+
+  // Run recursively over confined volumes
+  if (tVolume.confinedVolumes() != nullptr) {
+    for (const auto& volume : tVolume.confinedVolumes()->arrayObjects()) {
+      convert(gctx, *volume, cOptions, cState);
+    }
+  }
+}
```
Comment (AJPfleger) [inline Plugins/ActSVG/src/TrackingGeometrySvgConverter.cpp:95]:
```suggestion
}
```

File: Examples/Python/src/Svg.cpp [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+                                   &Writer::write));
+
+    auto c = py::class_<Writer::Config>(w, "Config").def(py::init<>());
+    ACTS_PYTHON_STRUCT_BEGIN(c, Writer::Config);
+    ACTS_PYTHON_MEMBER(outputDir);
+    ACTS_PYTHON_MEMBER(converterOptions);
+    ACTS_PYTHON_STRUCT_END();
+  }
+}
+}  // namespace Acts::Python
```
Comment (AJPfleger) [inline Examples/Python/src/Svg.cpp:116]:
```suggestion
}  // namespace Acts::Python
```

**Review by AJPfleger — CHANGES_REQUESTED** [review]

File: Examples/Io/CMakeLists.txt (hunk 2 above)
Comment (asalzburger) [reply Examples/Io/CMakeLists.txt:9]: Can't see the difference?

File: Examples/Python/src/SvgStub.cpp (hunk 4 above)
Comment (asalzburger) [reply Examples/Python/src/SvgStub.cpp:13]: Done

File: Examples/Io/CMakeLists.txt (hunk 2 above)
Comment (asalzburger) [reply Examples/Io/CMakeLists.txt:9]: I see it's a newline change, for Cmake files we can leave this I'd say

File: Examples/Python/src/JsonStub.cpp (hunk 3 above)
Comment (asalzburger) [reply Examples/Python/src/JsonStub.cpp:13]: Done.

File: thirdparty/actsvg/README.md (hunk 5 above)
Comment (asalzburger) [reply thirdparty/actsvg/README.md:5]: Also fixed VecMem.

File: Tests/UnitTests/Plugins/ActSVG/TrackingGeometrySvgConverterTests.cpp (hunk 6 above)
Comment (asalzburger) [reply Tests/UnitTests/Plugins/ActSVG/TrackingGeometrySvgConverterTests.cpp:61]: Fixed.

File: Tests/UnitTests/Plugins/ActSVG/LayerSvgConverterTests.cpp (hunk 7 above)
Comment (asalzburger) [reply Tests/UnitTests/Plugins/ActSVG/LayerSvgConverterTests.cpp:219]: Fixed.

File: Plugins/ActSVG/src/TrackingGeometrySvgConverter.cpp (hunk 8 above)
Comment (asalzburger) [reply Plugins/ActSVG/src/TrackingGeometrySvgConverter.cpp:95]: Fixed.

**Comment by asalzburger** [comment]
All comments should be addressed.

**Comment by asalzburger** [comment]
HI @AJPfleger - all comments should be addressed.

**Comment by asalzburger** [comment]
Review passed, was just overwritten by a revert of a file.

**Pull request merged**